### PR TITLE
fix: Enable redundantSelfInClosure and selfInPropertyInitialization - no ticket

### PR DESCRIPTION
**Pull request by agisilaos**
This pull request enhances our SwiftLint configuration by enabling two additional rules:

1. `redundant_self_in_closure`: This rule helps identify and prevent unnecessary use of `self` within closures, promoting cleaner and more concise code.

2. `self_in_property_initialization`: This rule encourages explicit use of `self` in property initializers, improving code clarity and reducing potential ambiguity.

Changes made:
- Added `redundant_self_in_closure` to the opt-in rules list
- Added `self_in_property_initialization` to the opt-in rules list

These additions will help maintain consistency across our codebase and align with Swift best practices.

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios-sync-engine/Source/SessionManager/APIVersionResolver.swift [hunk 1]
```diff
@@ -99,8 +99,7 @@ final class APIVersionResolver {
         } else if
             isDeveloperModeEnabled,
             let preferredAPIVersion = BackendInfo.preferredAPIVersion,
-            allBackendVersions.contains(preferredAPIVersion)
-        {
+            allBackendVersions.contains(preferredAPIVersion) {
             WireLogger.environment.info("resolving to preferred api version \(preferredAPIVersion.rawValue)")
             BackendInfo.apiVersion = preferredAPIVersion
         } else if let apiVersion = commonProductionVersions.max() {
```
Comment (markusfassbender) [inline wire-ios-sync-engine/Source/SessionManager/APIVersionResolver.swift]: For me the indentation of if and the block are difficult to read for the eye... maybe a blank line can help?

File: wire-ios-sync-engine/Source/Calling/MLSConferenceStaleParticipantsRemover.swift [hunk 2]
```diff
@@ -165,10 +165,12 @@ class MLSConferenceStaleParticipantsRemover: Subscriber {
                     guard let self else { return }
 
                     WaitingGroupTask(context: syncContext) { [self] in
+                        // swiftlint:disable redundant_self_in_closure
                         await self.remove(
                             client: clientID,
                             from: groupID
                         )
+                        // swiftlint:enable redundant_self_in_closure
```
Comment (markusfassbender) [inline wire-ios-sync-engine/Source/Calling/MLSConferenceStaleParticipantsRemover.swift]: These comments are not needed and even create a "superfluous_disable_command" warning.

File: wire-ios-data-model/Source/Model/Conversation/AssetCollection.swift [hunk 3]
```diff
@@ -91,20 +91,21 @@ public class AssetCollection: NSObject, ZMCollection {
...
         syncMOC.performGroupedBlock { [weak self] in
+            // swiftlint:disable redundant_self_in_closure
             guard let self, !self.tornDown else { return }
             guard let conversation = self.conversation,
                 let syncConversation = (try? syncMOC.existingObject(with: conversation.objectID)) as? ZMConversation else {
                 return
             }
-
-            let categorizedMessages: [ZMMessage] = AssetCollectionBatched.categorizedMessages(for: syncConversation, matchPairs: self.matchingCategories)
+            // swiftlint:enable redundant_self_in_closure
```
Comment (markusfassbender) [inline wire-ios-data-model/Source/Model/Conversation/AssetCollection.swift]: These comments are not needed and even create a `superfluous_disable_command` warning.

File: wire-ios-data-model/Source/Model/Conversation/AssetCollectionBatched.swift [hunk 4]
```diff
@@ -78,22 +78,24 @@ public class AssetCollectionBatched: NSObject, ZMCollection {
...
         syncMOC.performGroupedBlock { [weak self] in
+            // swiftlint:disable redundant_self_in_closure
             guard let self, !self.tornDown else { return }
             guard let conversation = self.conversation,
                   let syncConversation = (try? syncMOC.existingObject(with: conversation.objectID)) as? ZMConversation else {
                 return
             }
-            let allAssetMessages: [ZMAssetClientMessage] = self.unCategorizedMessages(for: syncConversation)
-            let allClientMessages: [ZMClientMessage] = self.unCategorizedMessages(for: syncConversation)
+            // swiftlint:enable redundant_self_in_closure
```
Comment (markusfassbender) [inline wire-ios-data-model/Source/Model/Conversation/AssetCollectionBatched.swift]: These comments are not needed and even create a `superfluous_disable_command` warning.

**Review by markusfassbender — APPROVED** [review]
Thanks for this cleanup!

**Review by netbe — COMMENTED** [review]
I am not convince we need the redundant_self_in_closure rule, maybe we can discuss it

**Review by markusfassbender — APPROVED** [review]

**Review by netbe — CHANGES_REQUESTED** [review]
remove redundant_self_in_closure or close this PR as it has been stale for a while

**Comment by agisilaos** [comment]
@netbe I'm closing this in favor of https://github.com/wireapp/wire-ios/pull/1959 Please review.

**Pull request closed without merging**